sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -247,6 +249,20 @@ class Analyzer(
     }
   }
 
+  /**
+   * Substitutes In values with an instance of [[InValues]].
+   */
+  object ResolveInValues extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
```

Comment:
-> `plan.resolveOperators`

Let us wait for https://github.com/apache/spark/pull/21822.